lib/ruby_smb/smb2/packet/ioctl_response.rb

```diff
@@ -17,7 +17,7 @@ class IoctlResponse < RubySMB::GenericPacket
         uint32        :output_count,        label: 'Output Count'
         uint32        :flags,               label: 'Flags'
         uint32        :reserved2,           label: 'Reserved Space'
-        string        :buffer,              label: 'Input Buffer'
+        string        :buffer,              label: 'Input Buffer',        length: -> { output_count }
 
```

Comment:
Also, it might be a good idea to add helper methods to extract input and output data (see [2.2.32 SMB2 IOCTL Response](https://msdn.microsoft.com/en-us/library/cc246548.aspx)):
```ruby
        def input_data
          to_binary_s[input_offset, input_count]
        end

        def output_data
          to_binary_s[output_offset, output_count]
        end
```